### PR TITLE
Fixes compatibility for iptables that don't support locking

### PR DESCRIPTION
**Pull request by janeczku**
Xtables locking is not supported on iptables before v1.4.20.
This causes go-iptables commands to fail with an error like this:

> iptables v1.4.8: unknown option `--wait'

Patch:
Adds check for `—Wait` command presence
Adds pseudo xtables lock via sync.mutex
Refactures version check (now done when init’ing IPTables object)

### CODE REVIEW

File: iptables/iptables.go [hunk 1]
```diff
@@ -40,7 +41,10 @@ func (e *Error) Error() string {
 }
 
 type IPTables struct {
-	path string
+	sync.Mutex
```
Comment (jonboulle) [inline iptables/iptables.go]: Convention is generally that the Mutex is listed before the members it protects - but in this case it's not used for any of the below - could you move it to the end of the struct instead?

**Comment by jonboulle** [comment]
Seems reasonable to me, /cc @eyakubovich

**Comment by eyakubovich** [comment]
The version check is LGTM. I don't know how much benefit the local mutex provides -- in lots of cases the contention is between multiple processes. Perhaps a better way would be to use an inter-process mechanism. Options include:

- SysV semaphores (semget)
- File lock (flock)
- Abstract unix socket

In earlier versions of iptables, xtables lock was implemented via a unix socket but it got changed early this year in favor of `flock` (http://git.netfilter.org/iptables/commit/?id=aa562a660d1555b13cffbac1e744033e91f82707).

I propose to use flock with the same filename, `/run/xtables.lock`, so that it is compatible with other, newer iptables versions that might be running on the host. I'm especially thinking of containers -- this way the same path must always be mounted in.

**Comment by janeczku** [comment]
@eyakubovich Interesting idea. I agree that using a mutex is probably just "better than nothing" approach. Looks pretty simpleI think the flock approach is especially great for containers. Mounting the lock file would mean thread-safety in both container and host level. Should be pretty easy to implement too.

**Comment by vcaputo** [comment]
@eyakubovich +1 for the flock approach at the common path.

**Comment by jonboulle** [comment]
@eyakubovich @vcaputo can one of you chase this up please?

**Comment by jonboulle** [comment]
Closing in favour of #12.

**Pull request closed without merging**